src/main/java/com/glencoesoftware/bioformats2raw/Converter.java

```diff
@@ -450,8 +466,15 @@ public void write(int series)
           throws FormatException, IOException, InterruptedException
   {
     String pathName = "/" + String.format(scaleFormatString, resolution - 1);
-    N5Reader n5 = new N5FSReader(
-        outputPath.resolve(pyramidName).toString());
+    final N5Reader n5;
```

Comment:
Maybe resolve the path once outside conditionals and construct with switch if we use a file type string above?

Reply:
_Hopefully_ there'll be some factory method one day, but for now :+1: